scripts/webknossos_annotation.py

```diff
@@ -0,0 +1,303 @@
+                        start_x, end_x = i*max_load, min((i+1)*max_load, subdat_size[-2])
+                        start_y, end_y = j*max_load, min((j+1)*max_load, subdat_size[-1])
+                        array[0, idx,  t + start_x: t + end_x, l + start_y: l + end_y] = subdat[start_x: end_x, start_y: end_y]
+                print('')
+                
+        
+    # Write OME-Zarr multiscale metadata
+    print('Write metadata')
+    multiscales = [{
+        'version': '0.4',
+        'axes': [
+            {"name": "z", "type": "space", "unit": "micrometer"},
+            {"name": "y", "type": "space", "unit": "micrometer"},
+            {"name": "x", "type": "space", "unit": "micrometer"}
+        ],
+        'datasets': [],
+        'type': 'jpeg2000',
+        'name': '',
+    }]
+    
+    if has_channel:
+        multiscales[0]['axes'].insert(0, {"name": "c", "type": "channel"})
+    for n in range(nblevel):
+        shape0 = omz['0'].shape[-2:]
+        shape = omz[str(n)].shape[-2:]
+        multiscales[0]['datasets'].append({})
+        level = multiscales[0]['datasets'][-1]
+        level["path"] = str(n)
+
+
+        level["coordinateTransformations"] = [
+            {
+                "type": "scale",
+                "scale": [1.0] * has_channel + [
+                    1.0, 
+                    (shape0[0]/shape[0])*vxh,
+                    (shape0[1]/shape[1])*vxw,
+                ]
+            },
+            {
+                "type": "translation",
+                "translation": [0.0] * has_channel + [
+                    0.0, 
+                    (shape0[0]/shape[0] - 1)*vxh*0.5,
+                    (shape0[1]/shape[1] - 1)*vxw*0.5,
+                ]
+            }
+        ]
+    multiscales[0]["coordinateTransformations"] = [
+        {
+            "scale": [1.0] * (3 + has_channel),
+            "type": "scale"
+        }
+    ]
+    omz.attrs["multiscales"] = multiscales
+
+
+
+def get_mask_name(level):
+    if level == 0:
+        return '1'
+    else:
+        return f'{2**level}-{2**level}-1'
+
+
+def ceildiv(x, y):
+    return int(math.ceil(x / y))
+
+
+def cal_distance(img):
+    m = img.shape[0]
+    for i in range(m):
+        cnt = np.sum(img[i, :])
+        if cnt > 0:
+            return i
+    return m  
+
+
+def find_borders(img):
+    if np.max(img) == 0:
+        return [-1, -1, -1, -1]
+    t = cal_distance(img)
+    b = cal_distance(img[::-1]) 
+    l = cal_distance(np.rot90(img, k=3)) 
+    r = cal_distance(np.rot90(img, k=1))
+
+    return [t, b, l, r]
+
+
+def contour_to_mask(mask, value):
+    h, w = mask.shape[:2]
+    kernel = np.ones((5, 5), np.uint8)
+    closed_mask = cv2.morphologyEx(mask, cv2.MORPH_CLOSE, kernel) 
+    single_mask = binary_fill_holes(closed_mask)
+
+    single_mask = np.where(single_mask, value, 0)
+    return single_mask
+
+
+def generate_mask(mask):
+    final_mask = np.zeros_like(mask).astype(np.uint8)
```

Comment:
What is this doing? Is it a good thing to process the data in a "conversion" script?